### PR TITLE
[Model] Update `CuGraphRelGraphConv` to use `pylibcugraphops=23.02`

### PR DESCRIPTION
**Pull request by tingyu66**
## Description

This PR update `CuGraphRelGraphConv` module to use `pylibcugraphops` 23.02.

With `pylibcugraphops` now offering aggregation autograd functions, we tidy up the source file to only include the nn.Module and make a few improvements.

Detailed changes include:
- support `apply_norm` option that enables normalized aggregation
- self loop weight is fused into the the weights `W` for better performance
- move `max_in_degree` to `forward()`, since it is a property of the graph not the model
- support full graph input (we only support sampled graph before)
- improve the test and update the example

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
Are there breaking changes?

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py [hunk 1]
```diff
@@ -214,87 +87,68 @@ def __init__(
         regularizer=None,
         num_bases=None,
         bias=True,
-        activation=None,
```
Comment (mufeili) [inline python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:217]: Is this a breaking change?

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py [hunk 2]
```diff
@@ -214,87 +87,68 @@ def __init__(
...
         self_loop=True,
         dropout=0.0,
-        layer_norm=False,
-        max_in_degree=None,
+        apply_norm=False,
     ):
         if has_pylibcugraphops is False:
             raise ModuleNotFoundError(
-                "dgl.nn.CuGraphRelGraphConv requires pylibcugraphops "
+                "dgl.nn.CuGraphRelGraphConv requires pylibcugraphops >= 23.02 "
```
Comment (mufeili) [inline python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:96]: Is there a way to check the version number of the package?

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py [hunk 3]
```diff
@@ -171,31 +49,26 @@ class CuGraphRelGraphConv(nn.Module):
         Default: ``None``.
     bias : bool, optional
         True if bias is added. Default: ``True``.
-    activation : callable, optional
-        Activation function. Default: ``None``.
     self_loop : bool, optional
         True to include self loop message. Default: ``True``.
     dropout : float, optional
         Dropout rate. Default: ``0.0``.
-    layer_norm : bool, optional
```
Comment (mufeili) [inline python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:180]: Is this a breaking change?

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py [hunk 4]
```diff
@@ -309,57 +163,58 @@ def forward(self, g, feat, etypes, norm=None):
             so any input of other integer types will be casted into int32,
             thus introducing some overhead. Pass in int32 tensors directly
             for best performance.
-        norm : torch.Tensor, optional
```
Comment (mufeili) [inline python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:312]: Is this replaced by `apply_norm=True`?

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py [hunk 5]
```diff
@@ -309,57 +163,58 @@ def forward(self, g, feat, etypes, norm=None):
...
-            A 1D tensor of edge norm value.  Shape: :math:`(|E|,)`.
+        max_in_degree : int, optional
+            Maximum in-degree of destination nodes. It is only effective when
+            :attr:`g` is a :class:`DGLBlock`, i.e., bipartite graph. When
```
Comment (mufeili) [inline python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:168]: Is this still only valid for `DGLBlock`?

File: tests/cugraph/cugraph-ops/test_cugraph_relgraphconv.py [hunk 6]
```diff
@@ -1,97 +1,77 @@
...
+
+import dgl
 import pytest
 import torch
-import dgl
-from dgl.nn import CuGraphRelGraphConv
-from dgl.nn import RelGraphConv
+from dgl.nn import CuGraphRelGraphConv, RelGraphConv
 
 # TODO(tingyu66): Re-enable the following tests after updating cuGraph CI image.
```
Comment (mufeili) [inline tests/cugraph/cugraph-ops/test_cugraph_relgraphconv.py:10]: Is this now re-enabled?

File: examples/advanced/cugraph/rgcn.py [hunk 7]
```diff
@@ -8,19 +8,20 @@
 code changes from the current `entity_sample.py` example.
```
Comment (mufeili) [inline examples/advanced/cugraph/rgcn.py:8]: Did you see similar performance numbers after running this script?

**Review by mufeili — COMMENTED** [review]
done a pass

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py (hunk 1 above)
Comment (tingyu66) [reply python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:217]: Yes. I saw the this comment from RelGraphConv:
https://github.com/dmlc/dgl/blob/76bb54044eb387e9e3009bc169e93d66aa004a74/python/dgl/nn/pytorch/conv/relgraphconv.py#L128-L129

I also prefer to apply activation outside of the conv layer.

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py (hunk 3 above)
Comment (tingyu66) [reply python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:180]: Yes, same as the "activation" argument above.

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py (hunk 4 above)
Comment (tingyu66) [reply python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:312]: Yes. This is an additional feature from CuGraphRelGraphConv: supporting the normalized aggregation presented in the RGCN paper.

Hence, users no longer need to compute the norm during training:
https://github.com/dmlc/dgl/blob/465828cdbc1adf570b4aa6c4922cc03bf7abdef2/examples/pytorch/rgcn/entity_sample.py#L60-L61

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py (hunk 5 above)
Comment (tingyu66) [reply python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py:169]: Yes

File: examples/advanced/cugraph/rgcn.py (hunk 7 above)
Comment (tingyu66) [reply examples/advanced/cugraph/rgcn.py:8]: Yes, performance is the same as before.

File: tests/cugraph/cugraph-ops/test_cugraph_relgraphconv.py (hunk 6 above)
Comment (tingyu66) [reply tests/cugraph/cugraph-ops/test_cugraph_relgraphconv.py:10]: Not yet, I will create a new image after our 23.02 release and update these pytest markers in a separate PR.

File: python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py (hunk 2 above)
Comment (tingyu66) [reply python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py]: https://github.com/rapidsai/dgl/blob/785e294b6e5a596df9c669eeb6ca56672a23d002/python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py#L11-L13

Any version <23.02 does not have `pylibcugraphops.torch.autograd` API, and thus will throw an exception here.

**Comment by tingyu66** [comment]
> Are there breaking changes?

Apologies for the late reply. The `norm` option is now replaced by `apply_norm`. Other breaking changes include the removal of `activation` and `layer_norm` options.

**Review by mufeili — APPROVED** [review]

**Comment by mufeili** [comment]
@dgl-bot

**Comment by mufeili** [comment]
@dgl-bot
